crates/pipeline_manager/src/db/mod.rs

```diff
@@ -29,14 +29,13 @@ pub(crate) mod storage;
 /// time, which determines the position of the project in the queue.
 pub(crate) struct ProjectDB {
     conn: Client,
+    // Used in dev mode for having an embedded Postgres DB live through the
+    // lifetime of the program.
+    #[cfg(feature = "pg-embed")]
+    #[allow(dead_code)] // It has to stay alive until ProjectDB is dropped.
+    pg_inst: Option<pg_embed::postgres::PgEmbed>,
```

Comment:
If we press ctrl+c on the manager in the console running as dev, and pg_inst is in a OnceCell, then drop() is not called, and so the Postgres process remains running (pg-embed does not stop it). So if you restart it after, it complains because it can't bring up the postgres instance again (it's still running the stale instance). So you have to manually go and kill postgres which is a bit annoying. If PgEmbed is in the struct then drop() gets called and Postgres is stopped.

with kill-9 this is still a problem even now: the postgres daemon is not cleaned up and we have to do it manually. But hopefully we won't need to kill-9 the process a lot anyways.

Reply:
I get that, but what I don't get is:

If OnceCell is dropped the PgEmbed instance should also be dropped right? Same as when we add the PgEmbed instance to the ProjectDB struct? What makes OnceCell not be dropped during sigterm?

Reply:
Because it's a static: 
> Static items do not call [drop](https://doc.rust-lang.org/reference/destructors.html) at the end of the program.
https://doc.rust-lang.org/reference/items/static-items.html

It has been discussed as expected behavior for OnceCell:
https://github.com/matklad/once_cell/issues/98

Reply:
Ah, missed the part about it being static. Thanks.